src/redux/task/actions.js

```diff
@@ -0,0 +1,179 @@
+        payload: order
+    }
+}
+
+const setPageLimit = (limit) => {
+    return {
+        type: SET_PAGE_LIMIT,
+        payload: limit
+    }
+}
+
+const setUpdateMode = (task, flag) => {
+    return {
+        type: SET_UPDATE_MODE,
+        payload: { task, flag }
+    }
+}
+
+const getTasks = (page) => {
+    return async(dispatch, getState) => {
+        const token = Cookies.get("token")
+        const { sortOrder, limit } = getState().task
+        const params = {
+            page,
+            limit,
+            sortOrder
+        }
+        dispatch(tasksRequest())
+        try {
+            const resp = await axios.get(
+                'http://localhost:3100/api/tasks', {
+                    params,
+                    headers: {
+                        Authorization: token
+                    }
+                }
+            )
+            const { tasks, page: currentPage, pagesCount: totalPages } = resp.data
+            dispatch(requestSuccess({ tasks, currentPage, totalPages }))
+        } catch (_error) {
+            dispatch(requestFailure())
+        }
+    }
+}
+
+const addTasks = (task) => {
+    return async(dispatch, getState) => {
+        const { text, date } = task
+        const token = Cookies.get("token")
+        const { currentPage } = getState().task
+        try {
+            await axios.post(
+                'http://localhost:3100/api/tasks', {
+                    text,
+                    date
+                }, {
+                    headers: {
+                        Authorization: token
+                    }
+                })
+            dispatch(getTasks(currentPage))
+        } catch (_error) {
+            dispatch(requestFailure())
+        }
+    }
+}
+
+const updateTask = (newTask) => {
+    return async(dispatch, getState) => {
+        const { _id, date, text } = newTask
+        const token = Cookies.get("token")
+        const { currentPage } = getState().task
+        dispatch(tasksRequest())
+        try {
+            await axios.patch(
+                `http://localhost:3100/api/tasks/${_id}`, {
+                    text,
+                    date
+                }, {
+                    headers: {
+                        Authorization: token
+                    }
+                })
+            dispatch(getTasks(currentPage))
+        } catch (_error) {
+            dispatch(requestFailure())
+        }
+    }
+}
+
+const deleteTask = (taskId) => {
+    return async(dispatch, getState) => {
+        const token = Cookies.get("token")
+        const { currentPage } = getState().task
+        dispatch(tasksRequest())
+        try {
+            await axios.delete(
+                `http://localhost:3100/api/tasks/${taskId}`, {
+                    headers: {
+                        Authorization: token
+                    }
+                })
+            dispatch(getTasks(currentPage))
+        } catch (_error) {
+            dispatch(requestFailure())
```

Comment:
There should be separate actions for add, update, and delete, for example on delete task success you can dispatch deleteTaskSuccess which will contain the newly created task, and update the state with that task. No need to make a separate API call. And onFailure it should dispatch something like deleteTaskFail which will update the error state of the application.